command/agent.go

```diff
@@ -507,21 +533,30 @@ func (c *AgentCommand) Run(args []string) int {
 	c.logGate.Flush()
 
 	// Write out the PID to the file now that server has successfully started
-	if err := c.storePidFile(config.PidFile); err != nil {
+	if err := c.storePidFile(agentConfig.PidFile); err != nil {
 		c.UI.Error(fmt.Sprintf("Error storing PID: %s", err))
 		return 1
 	}
 
 	defer func() {
-		if err := c.removePidFile(config.PidFile); err != nil {
+		if err := c.removePidFile(agentConfig.PidFile); err != nil {
 			c.UI.Error(fmt.Sprintf("Error deleting the PID file: %s", err))
 		}
 	}()
 
+	// Wait for the template to render
+	select {
+	case <-ctx.Done():
+	case <-unblockCh:
```

Comment:
Might be nice to rename `unblockCh` to something more descriptive, like `renderDoneCh`.

Reply:
I half agree with you here... I got this naming from Consul Template and at first I thought it odd naming, and would agree with your suggestion. After using it I came to agree with the naming though. I feel like `unblockCh` makes it clear that the purpose of the channel is to block execution somewhere until it closes. `renderDoneCh` is more descriptive that a render has taken place, but doesn't convey the purpose of that channel, which is to make execution wait until that render happens. 

That said, I'm not totally opposed to changing it, I just came around to agree with the naming after I looked at it long enough and used it. Let me know what you think

Reply:
That's fine by me.